### PR TITLE
Subscriber list for orders

### PR DESCRIPTION
**Pull request by marcofucci**
This adds the ability to specify a list of advisers to be subscribed to an order.

The 2 endpoints are:

**Get subscriber list**
GET `/v3/omis/order/<order-id>/subscriber-list`

Returns the list of advisers subscribed to an order:
```
[
    {
        "id": "e83a608e-84a4-11e6-ae22-56b6b6499611",
        "first_name": "Puck",
        "last_name": "Head",
        "dit_team": {
            "name": "CBBC North EAST",
            "id": "602b971d-5df6-e511-888e-e4115bead28a"
        }
    },
    {
        "id": "25628b23-c75b-4aef-b120-dac2d64c0696",
        "first_name": "",
        "last_name": "",
        "dit_team": null
    }
]
```

**Change subscriber list**
PUT `/v3/omis/order/<order-id>/subscriber-list`
with data:
```
[
    {
        "id": "e83a608e-84a4-11e6-ae22-56b6b6499611"
    },
    {
        "id": "25628b23-c75b-4aef-b120-dac2d64c0696"
    }
]
```

Changes the list to reflect the new one removing the advisers that are not in the list any more and adding new ones.

Documentation [here](https://documenter.getpostman.com/view/2403194/omis/6fZxPC3#2d9032ef-27de-bcc3-8666-bb0df771f1e4)

### CODE REVIEW

File: datahub/omis/order/test/test_views.py [hunk 1]
```diff
@@ -133,3 +137,154 @@ def test_not_found(self):
...
+
+        assert response.status_code == status.HTTP_200_OK
+        assert {adv['id'] for adv in response.json()} == {adv.id for adv in advisers}
+
+    def test_change_existing_list(self):
+        """
+        Test that calling PUT with a different list of advisers completely changes
+        the subscriber list.
+        It keeps the advisers passed in that were already on the list.
+        It removes the advisers what were on the list but not any more.
```
Comment (elcct) [inline datahub/omis/order/test/test_views.py]: Maybe this needs a little rewording

File: datahub/omis/order/models.py [hunk 2]
```diff
@@ -72,3 +72,26 @@ def save(self, *args, **kwargs):
...
             self.reference = self._calculate_reference()
         return super().save(*args, **kwargs)
+
+
+class OrderSubscriber(BaseModel):
+    """
+    A subscribed adviser receives notifications when new changes happen to an Order.
+    """
+
+    id = models.UUIDField(primary_key=True, db_index=True, default=uuid.uuid4)
```
Comment (reupen) [inline datahub/omis/order/models.py]: Any reason for specifying `db_index=True` here (other than it being done elsewhere)?

Comment (canni) [inline datahub/omis/order/models.py]: primary key implies implicit `UNIQUE` index in every DB, this just adds overhead

Comment (reupen) [inline datahub/omis/order/models.py]: Yes, that's why I asked, though looks like Django just ignores it.

Comment (marcofucci) [inline datahub/omis/order/models.py]: 👍  yes it was jut a lazy copy/paste :)

**Pull request merged**